rfcs/RFC-Upload-From-Dropbox.md

```diff
@@ -0,0 +1,358 @@
+| 400  | if the file type is invalid                                                                                      |
+
+#### Delete submission/{submission_id}/upload
+
+Cancels an existing upload job. Any data that has started to upload is removed, and the upload status is changed to
+cancel pending until the job has been cleared up.
+
+**Request:**
+
+| Parameter     | Description                           |
+| ------------- | ------------------------------------- |
+| submission_id | identifies the submission             |
+| dataset_id    | identifies the dataset being canceled |
+
+**Response:**
+
+| Key    | Description                                |
+| ------ | ------------------------------------------ |
+| status | Provides the current status of the upload. |
+
+**Error Responses:**
+
+| Code | Description                                                                                                             |
+| ---- | ----------------------------------------------------------------------------------------------------------------------- |
+| 401  | if dataset_id or submission_id does not exist, or if the user does not own the submission or upload in-progress upload. |
+| 400  | if the parameters supplied are invalid.                                                                                 |
+
+#### Upload Status and State
+
+The API documentation returns a _status_ in response. This status informs the user of the current state of the upload.
+An upload follows this state diagram:
+
+![Upload States](https://app.lucidchart.com/publicSegments/view/5fe0858f-eb93-4b64-9842-6f25cad371f3/image.png)
+Figure 2: Upload State Diagram
+
+There are several different states that an upload can be in. The table below defines them.
+
+| #   | states         | Description                                                                                 |
+| --- | -------------- | ------------------------------------------------------------------------------------------- |
+| S0  | Start          | The upload is never actually in this state.                                                 |
+| S1  | Waiting        | The upload is enqueued, and waiting for computing resources.                                |
+| S2  | uploading      | The file is actively being uploaded.                                                        |
+| S3  | Completed      | The upload was completed successfully.                                                      |
+| S4  | Failed         | The upload has failed. A new link or file must be uploaded. Any upload progress is deleted. |
+| S5  | Cancel Pending | The upload is in the process of being canceled                                              |
+| S6  | Canceled       | The upload has been canceled. Any upload progress is deleted.                               |
+
+These states will be stores in the upload table until the upload is complete.
```

Comment:
s/stores/stored

If the states are stored until the upload is Complete, then is it possible to see "Complete" in the table?
Possibly, we need to keep the Complete status in the table until the upload submission has moved onto the next stage in the pipeline.

Reply:
How long do we store "Canceled", "Failed".
Should there be a mechanism to clear them?

Reply:
I've been talking with Marcus about extending the dataset status in the Database to use the states. If we do that, then the state would transition to validating after complete. Once the validation is complete and the file moves to the primary bucket we should clear this state.